float64 x
float64 y
float64 z
float64 ro
float64 pi
float64 ya
